cmd/osbuild-tests/main.go

```diff
@@ -0,0 +1,249 @@
+		log.Fatalf("Unexpected compose result: %s", status)
+	}
+}
+
+func startCompose(name, outputType string) uuid.UUID {
+	var reply struct {
+		BuildID uuid.UUID `json:"build_id"`
+		Status  bool      `json:"status"`
+	}
+	rawReply := runCommand(false, "compose", "start", name, outputType)
+	err := json.Unmarshal(rawReply, &reply)
+	if err != nil {
+		log.Fatalf("Unexpected reply: " + err.Error())
+	}
+	if reply.Status != true {
+		log.Fatalf("Unexpected status %v", reply.Status)
+	}
+
+	return reply.BuildID
+}
+
+func deleteCompose(id uuid.UUID) {
+	type deleteUUID struct {
+		ID     uuid.UUID `json:"uuid"`
+		Status bool      `json:"status"`
+	}
+	var reply struct {
+		IDs    []deleteUUID  `json:"uuids"`
+		Errors []interface{} `json:"errors"`
+	}
+	rawReply := runCommand(false, "compose", "delete", id.String())
+	err := json.Unmarshal(rawReply, &reply)
+	if err != nil {
+		log.Fatalf("Unexpected reply: " + err.Error())
+	}
+	if len(reply.Errors) != 0 {
+		log.Fatalf("Unexpected errors")
+	}
+	if len(reply.IDs) != 1 {
+		log.Fatalf("Unexpected number of UUIDs returned: %d", len(reply.IDs))
+	}
+	if reply.IDs[0].Status != true {
+		log.Fatalf("Unexpected status %v", reply.IDs[0].Status)
+	}
+}
+
+func waitForCompose(uuid uuid.UUID) string {
+	for {
+		status := getComposeStatus(true, uuid)
+		if status == "FINISHED" || status == "FAILED" {
+			return status
+		}
+		time.Sleep(time.Second)
+	}
+}
+
+func getComposeStatus(quiet bool, uuid uuid.UUID) string {
+	var reply struct {
+		QueueStatus string `json:"queue_status"`
+	}
+	rawReply := runCommand(quiet, "compose", "info", uuid.String())
+	err := json.Unmarshal(rawReply, &reply)
+	if err != nil {
+		log.Fatalf("Unexpected reply: " + err.Error())
+	}
+
+	return reply.QueueStatus
+}
+
+func pushBlueprint(bp *blueprint.Blueprint) {
+	tmpfile, err := ioutil.TempFile("", "osbuild-test-")
+	if err != nil {
+		log.Fatalf("Could not create temporary file: " + err.Error())
+	}
+	defer os.Remove(tmpfile.Name())
+
+	err = toml.NewEncoder(tmpfile).Encode(bp)
+	if err != nil {
+		log.Fatalf("Could not marshapl blueprint TOML: " + err.Error())
+	}
+	if err := tmpfile.Close(); err != nil {
+		log.Fatalf("Could not close toml file: " + err.Error())
+	}
+
+	var reply struct {
+		Status bool `json:"status"`
+	}
+	rawReply := runCommand(false, "blueprints", "push", tmpfile.Name())
+	err = json.Unmarshal(rawReply, &reply)
+	if err != nil {
+		log.Fatalf("Unexpected reply: " + err.Error())
+	}
+	if reply.Status != true {
+		log.Fatalf("Unexpected status %v", reply.Status)
+	}
+}
+
+func deleteBlueprint(bp *blueprint.Blueprint) {
+	var reply struct {
+		Status bool `json:"status"`
+	}
+	rawReply := runCommand(false, "blueprints", "delete", bp.Name)
+	err := json.Unmarshal(rawReply, &reply)
+	if err != nil {
+		log.Fatalf("Unexpected reply: " + err.Error())
+	}
+	if reply.Status != true {
+		log.Fatalf("Unexpected status %v", reply.Status)
+	}
+}
+
+func runCommand(quiet bool, command ...string) json.RawMessage {
```

Comment:
Do you mind naming this `runComposerCLI` or something more descriptive?

Reply:
Good point! Done.